### PR TITLE
Send `NoOrchestrators` event

### PR DESCRIPTION
**Pull request by darkdarkdragon**
**What does this pull request do? Explain your changes. (required)**
Send `NoOrchestrators` event in case source segment appeared
but there is not orchestrators selected

**Specific updates (required)**

- 
- 
- 

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

**Checklist:**

- [ ] README and other documentation updated
- [x] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

### CODE REVIEW

File: monitor/events.go [hunk 1]
```diff
@@ -234,6 +235,16 @@ func LogSegmentTranscodeFailed(subType string, nonce, seqNo uint64, err error) {
...
 	sendPost("SegmentTranscodeFailed", nonce, props)
+
+	if subType == "NoOrchestrators" {
+		// this will allow us to find out at what stages of stream "NoOrchestrators" errors occur
+		props2 := map[string]interface{}{
+			"subType": fmt.Sprintf("NoOrchestrators-seq-%04d", seqNo),
+			"reason":  err.Error(),
+			"seqNo":   seqNo,
+		}
+		sendPost("SegmentTranscodeFailed", nonce, props2)
```
Comment (j0sh) [inline monitor/events.go]: It seems like we'd be sending SegmentTranscodeFailed twice? Once with `props` and again with `props2` ?

Comment (darkdarkdragon) [reply monitor/events.go]: Yes, but they're get sent with different `subType` - second one differentiate events by SeqNo: `NoOrchestrators-seq-%04d`. They will be shown in metrics separately, so we will be able to see when in the stream these errors took place.

Comment (j0sh) [reply monitor/events.go]: Don't we also send the sequence number in the `seqNo` field along with the subtype?

Is this temporary until Prometheus? Will we have a better story for drilling down into this type of event as opposed to double-counting errors?

Can you confirm this is the source of the discrepancy in upload behavior by running the test harness with this patch?

Comment (darkdarkdragon) [reply monitor/events.go]: > Don't we also send the sequence number in the seqNo field along with the subtype?

Yes, we do. But sending it two times allow me to show separate entry containing `seqNo` in metrics UI without changing metrics server code.

image



> Is this temporary until Prometheus? 

All code related to own metrics should be removed eventually, when we will be satisfied with Prometheus based metrics. 

> Will we have a better story for drilling down into this type of event as opposed to double-counting errors?

I'm working on that.

> Can you confirm this is the source of the discrepancy in upload behavior by running the test harness with this patch?

Wasn't able to reproduced, but I haven't tried too hard.

Comment (j0sh) [reply monitor/events.go]: > Wasn't able to reproduced, but I haven't tried too hard.

My guess is that it'd be easier to track down this specific behavior by:

* Drilling down on errors for a specific stream (maybe we can avoid this if load is light)
* Viewing seqNo for all errors, in addition to NoOrchestrators.
* Seeing if we're losing the O in relation to some other error (seqNo should closely follow between errors).

The approach here seems a little ad-hoc, especially if it hasn't already helped to give a sense of *why* the errors are happening. If it's not worth modifying the metrics server code then maybe we should focus on getting this type of visibility through Prometheus, especially if all this is going to be removed soon anyway.

**Comment by darkdarkdragon** [comment]
@j0sh I've removed error duplication.

**Review by j0sh — APPROVED** [review]
🚢

**Pull request merged**